### ISSUE TITLE
no reply memory leak

### ISSUE THREAD

**Issue by brandonpsmith** [arobson/rabbot#8]
Rabbot is subscribing to a postal channel with the requestId and not unsubscribing until the reply is sent. If the reply is not sent the subscriptions just build up. Watch the memory usage in the example below.  Should there be a timeout on the reply? idk

```javascript
'use strict';

const rabbit = require('rabbot');
const topology = {
  connection: {
    server: 'localhost',
  },

  exchanges: [{
    name: 'test.ex',
    type: 'direct',
  }],

  queues: [{
    name: 'test.q',
    noAck: true,
  }],

  bindings: [{
    exchange: 'test.ex',
    target: 'test.q',
    keys: 'test',
  }],
};

rabbit.configure(topology).done( function() {
  rabbit.handle('test', (message) => {
    console.log(`received ${message.body}`)
    //message.reply(`${message.body}`);
  });

  rabbit.startSubscription('test.q');

  (function send(num) {
    const options = {
      body: num.toString(),
      type: 'test',
    };

    rabbit.request('test.ex', options).then((message) => {
      //never gets called
      console.log(`replied ${message.body}`)
    });

    setTimeout(send.bind(undefined, num+1), 0);
  }(1));
});

```

**Comment by arobson** [comment]
@brandonpsmith - thanks for pointing this out, I have put `replyTimeout` in place and updated the README to explain it. Basically, if a publishTimeout exists and no replyTimeout was specified, it just doubles that for the timeout. Let me know what you think, the code is available in NPM as version 1.0.0-9.

**Comment by brandonpsmith** [comment]
Looks good. If I find anything else. I'll let you know.
